### PR TITLE
Adds tests for Path helper.

### PR DESCRIPTION
**Pull request by dbrumann**
I wanted to make sure I understand how concatenate on Path works for my
PR for #513 and maybe the tests are valuable to others as well.

### CODE REVIEW

File: tests/PathTest.php [hunk 1]
```diff
@@ -22,4 +22,27 @@ public function testConcatenateOnWindows()
...
+    /**
+     * @dataProvider providePathsForConcatenation
+     */
+    public function testConcatenate($part1, $part2, $expectedPath)
+    {
+        $path = new Path('');
+
+        $actualPath = $path->concatenate([$part1, $part2]);
+
+        $this->assertEquals($expectedPath, $actualPath);
```
Comment (xabbuh) [inline tests/PathTest.php]: `assertSame()`?

Comment (dbrumann) [reply tests/PathTest.php]: I can change it, but I chose equals to keep it consistent with the existing test.

**Comment by dbrumann** [comment]
Closed in favor of https://github.com/symfony/flex/pull/551

**Comment by dbrumann** [comment]
Dang it, wrong PR 🙈 reopened

**Comment by nicolas-grekas** [comment]
Thank you @dbrumann, eventually :)

**Pull request merged**